### ISSUE TITLE
Popup is stretched to the entire window in the android app home assistant

### ISSUE THREAD

**Issue by andrey170183** [thomasloven/hass-browser_mod#606]
My Home Assistant version: 2023.7.2

**What I am doing:**
Updated browser_mod to version 2.3.0

**What I expected to happen:**
Open in a small window as before before the update

**What happened instead:**
popup opened for the whole window in the android app home assistant

**Minimal** steps to reproduce:
Just updated browser_mod to version 2.3.0
![Screenshot_20230715-190309149](https://github.com/thomasloven/hass-browser_mod/assets/42129962/314be7a0-5aae-48e0-a90e-3a7ad9d88a61)

**Comment by WW1983** [comment]
Same on iPhone

**Comment by K-RAD** [comment]
Same here (Android 13)

**Comment by limitless00net** [comment]
same here..

**Comment by rob-oravec** [comment]
Me too - iPhone. Have tried pretty much everything (I think) - style changes, size changes as per documentation, theme changes, clear cache.

**Comment by Namsikjo** [comment]
I'm having the same problem too. (iPhone14Pro iOS17, iPhone11Pro iOS16)

**Comment by Lux315** [comment]
Got this issue aswell.

Seems to be the same
issue like here: https://github.com/thomasloven/hass-browser_mod/issues/617

@thomasloven Could you please take a look?

**Comment by suheilpurewal** [comment]
same issue for me on Android (Pixel 7 pro) & iPhone (14 pro max). Seems to be working fine on iPad & PC/Mac.

**Comment by ASNNetworks** [comment]
> same issue for me on Android (Pixel 7 pro) & iPhone (14 pro max). Seems to be working fine on iPad & PC/Mac.

True, it affects popups on mobiles only. Basically devices that show the default more-info dialogs in a window are unaffected. Devices that show the more-info dialogs in a fullscreen manner (the core dialogs) now also show browser_mod popups in fullscreen (with a bugged out header).

**Comment by roderik-maker** [comment]
A temporary and not optimal fix, is to install card_mod and add the following to your popup cards:
(For IOS, haven't yet tested on android)

```
card_mod:
  style: |
    ha-dialog {
      --mdc-dialog-min-height: 10px !important;
    }
```
This returns the popups to a small size however, they are now at the bottom of the screen. I haven't yet figured out how to move them back up, but at least you don't have to quit the app to close a popup (This should help until a proper fix is implemented).

Example of full popup card:

```
type: custom:popup-card
title: Custom popup
dismissable: true
card:
  type: markdown
  content: This replaces the more-info dialog
right_button: right
left_button: left
entity: input_button.battery
card_mod:
  style: |
    ha-dialog {
      --mdc-dialog-min-height: 10px !important;
    }
```

**Comment by jbrandek** [comment]
Ive bee able to fix it the problem is in browser_mod.js and browser_mod.js.gz on media querry 450px
you should change it to this:

@media all and (max-width: 450px), all and (max-height: 500px) {
        ha-dialog {
          --mdc-dialog-min-width: 97vw;
          --mdc-dialog-max-width: 97vw;
          --mdc-dialog-min-height: 10%;
          --mdc-dialog-max-height: 100%;
        }
then delete cache

i live the files if someone does not know how to edit them, just delete the ones from custom_components/browser_mod/

and replace this 2

dont know how to edit in github if someone knows please upload them so they are kept in next update
[browser_mod.zip](https://github.com/thomasloven/hass-browser_mod/files/12317580/browser_mod.zip)

**Comment by ve1koz111** [comment]
#617 Fix at the bottom

**Comment by ve1koz111** [comment]
@smeen89 https://github.com/thomasloven/hass-browser_mod/issues/617#issuecomment-1676580513

**Comment by smeen89** [comment]
> @smeen89 [#617 (comment)](https://github.com/thomasloven/hass-browser_mod/issues/617#issuecomment-1676580513)

You beautiful magician! This has been bugging me for so long and now it works! Great job!

**Comment by ASNNetworks** [comment]
@ve1koz111 thank you so much for your fix. I tried it and works just like it should :) Hopefully it will be merged with the official.

**Comment by JoaoPedro13** [comment]
this is still an issue, replacing the files did not do the trick for me

**Comment by thomasloven** [comment]
Could someone please post a screenshot of what the normal non-browser_mod more-info dialog popup looks like on the android app?

**Comment by Dino-Tech** [comment]
@thomasloven 
![Screenshot_20240219-171224](https://github.com/thomasloven/hass-browser_mod/assets/38272439/2bb5a760-5abf-43b4-bcc7-26a79c3de515)
![Screenshot_20240219-171349](https://github.com/thomasloven/hass-browser_mod/assets/38272439/b79cd3d8-9e30-448d-a4a6-373244656080)
![Screenshot_20240219-171422](https://github.com/thomasloven/hass-browser_mod/assets/38272439/ee4b4016-259c-4318-a945-a1bc8e0b0f8b)
![Screenshot_20240219-171522](https://github.com/thomasloven/hass-browser_mod/assets/38272439/c996c89d-89b3-4c6f-81ce-d53a2631e22d)

**Comment by emilio6479** [comment]
again popup opened for the whole window in the android app home assistant.
help pls

**Comment by Sneakybun** [comment]
Had the same issue, found the "height" is defined as 100%, change it to "auto" solves the issue on android, add "--ha-dialog-border-radius:" if you want a radius on the popup.

      browser_mod:
        service: browser_mod.popup
        data:
          card_mod:
            style:
              ha-dialog$: |
                div.mdc-dialog div.mdc-dialog__container {
                  height: auto;
                }

**Comment by dcapslock** [comment]
Noting that v2.3.4 and onwards has `classic` style dialog size.

**Comment by dcapslock** [comment]
v2.3.8 now released with `classic` size available,

**Comment by lcsiew** [comment]
i found the culprit is due to the .mdc_dialog__surface 's min_height
<img width="1758" height="809" alt="Image" src="https://github.com/user-attachments/assets/03a46eb1-6df5-400a-87e4-284c7e7cd3b5" />

if i set it to auto, the popup's height will be set according to its content's height
<img width="1634" height="622" alt="Image" src="https://github.com/user-attachments/assets/177be543-9b11-4cd0-8860-c86cfe7b0754" />

however i cannot make this card_mod to work. please assist if you know how to make it work
```
    card_mod:
      style: 
        ha-dialog$: |         
          @media (max-width: 450px) {
            div.mdc-dialog__surface {
              min-height: auto !important; 
            }
          }
```

**Comment by dcapslock** [comment]
Update to Home Assistant 2025.10.2. This sizing broke in 2025.10.0
